Products/zms/standard.py

```diff
@@ -1320,8 +1320,19 @@ def compareDate(t0, t1):
   @returns: A negative number if date t0 is before t1, zero if they are equal, or positive if t0 is after t1.
   @rtype: C{int}
   """
-  mt0 = time.mktime(stripDateTime(getDateTime(t0)))
-  mt1 = time.mktime(stripDateTime(getDateTime(t1)))
+  
+  if isinstance(t0, time.struct_time):

```

Comment:
def make_time(t):
  if isinstance(t, time.struct_time):
    # Normalize tm_dst-variable (DST: Daylight Saving Time) 
    # in struct_time not to trigger mktime-error 
    dt = DateTime(format_datetime_iso(t))
  else:
    dt = getDateTime(t)
mt0 = time.mktime(stripDateTime(dt))

mt0 = make_time(t0)
mt1 = make_time(t1)